e2e/utils/waits-utils.ts

```diff
@@ -5,10 +5,14 @@ import Container from 'app/container';
 import { makeDateTimeStr } from './str-utils';
 import { getAttrValue } from './element-utils';
 
-export const waitForFn = async (fn: () => any, interval = 2000, timeout = 10000): Promise<boolean> => {
+export const waitForFn = async (
+  fn: (() => Promise<boolean>) | (() => boolean),
+  interval = 2000,
+  timeout = 10000
+): Promise<boolean> => {
   const start = Date.now();
   while (Date.now() < start + timeout) {
-    if (fn()) {
+    if (await fn()) {
```

Comment:
TIL you can `await` non-Promises. "[await converts the value to a resolved Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#:~:text=await%20converts%20the%20value%20to%20a%20resolved%20Promise)".